torch/csrc/jit/codegen/cuda/fusion_segmenter.h

```diff
@@ -295,6 +298,110 @@ class TORCH_CUDA_CU_API SegmentedFusion {
   //! Cleanup function to be call at the end of fusion
   //!  segment pass
   void finalize();
+
+  //! Utility to give unique name for each segmented fusion
+  static size_t segmentedFusionName() {
+    static size_t counter = 0;
+    return counter++;
+  }
+};
+
+//! An utility class to compute and maintain the "producers of"
+//!   relationship in a segmented graph. Space heavy and should
+//!   avoid use on very large graphs.
+//!
+//!  Currently trying to move as far as possible with only a
+//!   producer map, without transposing it to make a consumer map.
+//!  Making it NonCopyable because we should never need to
+//!   copy an instance of this class.
+//!  TODO: Space efficiency of this class will be important,
+//!        because we need it in the pre-merging of segmentedGroups,
+//!        currently O(n^2). O(nlogn) would be a reasonable
+//!        goal to achieve.
+class GroupDependencyAnalysis : public NonCopyable {
```

Comment:
This doesn't need to be exposed in a header. We can leave it there, but my understanding is it could simply be in the .cpp under an anonymous namespace.